skrub/_table_vectorizer.py

```diff
@@ -613,6 +613,14 @@ def fit_transform(self, X: ArrayLike, y: ArrayLike = None) -> ArrayLike:
         # in numerical columns for instance.
         X = _replace_false_missing(X)
 
+        ###
+        # We need to check for duplicate column names.
+        # It is checked by comparing the number of unique values
+        # to the length of the column names
+        if len(set(X.columns)) != len(X.columns):
+            raise AssertionError("Duplicate column names in the dataframe")
```

Comment:
Can you add to the error message the values of X.columns, so as to add the user to diagnose the problem. Something like the suggestion below:
```suggestion
            raise AssertionError("Duplicate column names in the dataframe."
                                 f"The column names are {X.columns}")
```